### PR TITLE
Feat/add reuse publishing

### PR DESCRIPTION
**Pull request by Jorek57**

### CODE REVIEW

File: udata_front/theme/gouvfr/assets/js/components/dataset/CardSM.vue
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/components/dataset/CardSM.vue:1]: remove and use CardLG in this PR (the current style will come with #419)

File: udata_front/theme/gouvfr/assets/js/components/Form/Stepper/StepperReuse.stories.js
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/components/Form/Stepper/StepperReuse.stories.js:1]: Is this file necessary ? I think it could be a simple `steps` array in `ReusePublishingForm`

File: udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/ReusePublishingForm.vue [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    reuse.value = await createReuse(newReuse);
+    moveToNextStep = true;
+  } catch (e) {
+    errors.value.push(e.message);
+  }
+  try {
+    const resp = await uploadLogo(reuse.value.id, newFile.value[0]);
+    reuse.value.image = resp.data.image
+  } catch (e) {
+    errors.value.push("Failed to upload logo, you can upload it again in your management panel");
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/ReusePublishingForm.vue:116]: Missing translation

File: udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/ReusePublishingForm.vue [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  const datasets = newReuse.datasets;
+  try {
+    reuse.value = await updateReuse(newReuse);
+    reuse.value.datasets = datasets;
+    moveToNextStep = true;
+  } catch (e) {
+    errors.value.push(e.message);
+  }
+  if (moveToNextStep) {
+    moveToStep(2)
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/ReusePublishingForm.vue:135]:
```suggestion
    moveToStep(2);
```

File: udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/ReusePublishingForm.vue [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  }
+};
+
+async function updateReuseDatas(newReuse: Reuse) {
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/ReusePublishingForm.vue:139]:
```suggestion
async function updateReuseData(newReuse: Reuse) {
```

File: udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/Step1DescribeReuse.vue [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+            {{ $t('Help') }}
+          </template>
+          <AccordionGroup>
+            <Accordion
+              :title= "$t('Naming your reuse')"
+              :id="nameReuseAccordionId"
+              :state="state.title"
+            >
+              <p class="fr-m-0">
+                {{ $t("Prefer a title that allows understanding of the use made of the data rather than the name of the website or application (\"Search Engine for Company Agreements\" rather than \"\" for example).") }}
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/Step1DescribeReuse.vue:21]: add `useI18n` ans use `t` instead

File: udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/Step1DescribeReuse.vue [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+      page: "",
+      slug: "",
+      uri: "",
+      logo_thumbnail: userValue.avatar_thumbnail || ""
+    });
+  }
+}
+
+onMounted(async () => {
+  await fetchUser()
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/Step1DescribeReuse.vue:435]:
```suggestion
  await fetchUser();
```

File: udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/Step1DescribeReuse.vue [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+});
+
+const submit = () => {
+  validateRequiredRules().then(valid => {
+    if(valid) {
+      if (toValue(userOrganization) !== "user") {
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/Step1DescribeReuse.vue:441]: I think the `user` case is never treated

**Comment by Jorek57** [comment]
PR redone here [#446](https://github.com/datagouv/udata-front/pull/446)
